simulator/src/main/java/com/hedera/block/simulator/generator/BlockAsDirBlockStreamManager.java

```diff
@@ -0,0 +1,165 @@
+    @Override
+    public Block getNextBlock() {
+        Block nextBlock = blocks.get(currentBlockIndex);
+        currentBlockIndex++;
+        lastGivenBlockNumber++;
+        if (currentBlockIndex >= blocks.size()) {
+            currentBlockIndex = 0;
+        }
+        return nextBlock;
+    }
+
+    private void loadBlocks() throws IOException, ParseException {
+        Path rootPath = Path.of(rootFolder);
+
+        try (Stream<Path> blockDirs = Files.list(rootPath).filter(Files::isDirectory)) {
+            List<Path> sortedBlockDirs =
+                    blockDirs.sorted(Comparator.comparing(Path::getFileName)).toList();
+
+            for (Path blockDirPath : sortedBlockDirs) {
+                List<BlockItem> parsedBlockItems = new ArrayList<>();
+
+                try (Stream<Path> blockItems =
+                        Files.list(blockDirPath).filter(Files::isRegularFile)) {
+                    List<Path> sortedBlockItems =
+                            blockItems
+                                    .sorted(
+                                            Comparator.comparing(
+                                                    BlockAsDirBlockStreamManager
+                                                            ::extractNumberFromPath))
+                                    .toList();
+
+                    for (Path pathBlockItem : sortedBlockItems) {
+                        byte[] blockItemBytes = readBlockItemBytes(pathBlockItem);
+                        // if null means the file is not a block item and we can skip the file.
+                        if (blockItemBytes == null) {
+                            continue;
+                        }
+                        BlockItem blockItem = BlockItem.PROTOBUF.parse(Bytes.wrap(blockItemBytes));
+                        parsedBlockItems.add(blockItem);
+                    }
+                }
+
+                blocks.add(Block.newBuilder().items(parsedBlockItems).build());
+                LOGGER.log(DEBUG, "Loaded block: " + blockDirPath);
+            }
+        }
+    }
+
+    private byte[] readBlockItemBytes(Path pathBlockItem) throws IOException {
+        if (pathBlockItem.toString().endsWith(".gz")) {
+            return Utils.readGzFile(pathBlockItem);
+        } else if (pathBlockItem.toString().endsWith(".blk")) {
+            return Files.readAllBytes(pathBlockItem);
+        } else {
+            return null;
+        }
+    }
+
+    // Method to extract the numeric part of the filename from a Path object
+    // Returns -1 if the filename is not a valid number
+    private static int extractNumberFromPath(Path path) {
+        String filename = path.getFileName().toString();
+        String numPart = filename.split("\\.")[0]; // Get the part before the first dot
+        try {
+            return Integer.parseInt(numPart);
+        } catch (NumberFormatException e) {
+            return -1; // Return -1 if parsing fails
```

Comment:
If this fails then it always ends up first in the sorted block items? Shouldn't we fails since we know there's a filename that needs to be repaired, etc?

Reply:
This is more for files that are not part of the block, we will end up ignoring the files, but their filenames cannot be parsed to number.